src/lib/core/fiber.c

```diff
@@ -258,11 +258,19 @@ pthread_t main_thread_id;
 static size_t page_size;
 static int stack_direction;
 
+#if FIBER_STACK_SIZE < 524288
+/*
+ * See the rationale for the minimal fiber stack size in
+ * https://github.com/tarantool/tarantool/issues/3418.
+ */
+#error "Minimal fiber stack size is 512Kb"
+#endif
+
 enum {
 	/* The minimum allowable fiber stack size in bytes */
 	FIBER_STACK_SIZE_MINIMAL = 16384,
 	/* Default fiber stack size in bytes */
-	FIBER_STACK_SIZE_DEFAULT = 524288,
+	FIBER_STACK_SIZE_DEFAULT = FIBER_STACK_SIZE,
```

Comment:
Nit: You could define `FIBER_STACK_SIZE_DEFAULT` right in cmake, instead of `FIBER_STACK_SIZE`.

Reply:
Fixed, thanks!

```diff
diff --git a/cmake/SetFiberStackSize.cmake b/cmake/SetFiberStackSize.cmake
index 58aa7f751..78404af02 100644
--- a/cmake/SetFiberStackSize.cmake
+++ b/cmake/SetFiberStackSize.cmake
@@ -31,14 +31,14 @@ math(EXPR FIBER_STACK_SIZE_IN_BYTES
 # https://github.com/tarantool/tarantool/issues/3418.
 if(FIBER_STACK_SIZE_IN_BYTES LESS 524288)
     message(FATAL_ERROR "[SetFiberStackSize] Minimal fiber stack size is 512Kb,"
-        " but ${FIBER_STACK_SIZE} is given")
+        " but ${FIBER_STACK_SIZE} is given as a default")
 else()
-    message(STATUS "[SetFiberStackSize] Fiber stack size: ${FIBER_STACK_SIZE}"
-        " (adjusted to ${FIBER_STACK_SIZE_IN_BYTES} bytes)")
+    message(STATUS "[SetFiberStackSize] Default fiber stack size: "
+        " ${FIBER_STACK_SIZE} (adjusted to ${FIBER_STACK_SIZE_IN_BYTES} bytes)")
 endif()
 
 # Propagate the value to the sources.
-add_definitions(-DFIBER_STACK_SIZE=${FIBER_STACK_SIZE_IN_BYTES})
+add_definitions(-DFIBER_STACK_SIZE_DEFAULT=${FIBER_STACK_SIZE_IN_BYTES})
 
 # XXX: Unset variables to avoid spoiliing CMake environment.
 unset(FIBER_STACK_SIZE_IN_BYTES)
diff --git a/src/lib/core/fiber.c b/src/lib/core/fiber.c
index cb0de871a..958e4a02f 100644
--- a/src/lib/core/fiber.c
+++ b/src/lib/core/fiber.c
@@ -258,15 +258,13 @@ pthread_t main_thread_id;
 static size_t page_size;
 static int stack_direction;
 
-#ifndef FIBER_STACK_SIZE
-#error "Fiber stack size is not set"
+#ifndef FIBER_STACK_SIZE_DEFAULT
+#error "Default fiber stack size is not set"
 #endif
 
 enum {
        /* The minimum allowable fiber stack size in bytes */
        FIBER_STACK_SIZE_MINIMAL = 16384,
-       /* Default fiber stack size in bytes */
-       FIBER_STACK_SIZE_DEFAULT = FIBER_STACK_SIZE,
        /* Stack size watermark in bytes. */
        FIBER_STACK_SIZE_WATERMARK = 65536,
 };
```